### ISSUE TITLE
`--show-errors` is not enabled by default and does not generate valid JSON

### ISSUE THREAD

**Issue by dylon** [lfortran/lfortran#5525]
`--show-errors` is not enabled by default but should be because it is required for the LSP integration. To enable it, you must install rapidjson and then pass the flag `-DWITH_JSON=yes` to cmake.

Also, something has changed that now makes it respond with invalid JSON. If I introduce an error to `tests/function_call1.f90` and call `--show-errors` on it, I get the following:

```shell
./src/bin/lfortran --show-errors tests/function_call1.f90
```

```json
{"uri":"uri","diagnostics":["source":"lpyth","range":{"start":{"line":20,"character":1},"end":{"line":20,"character":5}},"message":"Statement or Declaration expected inside program, found Variable name","severity":0}]}
```

This is invalid for the following reason:

```
parse error: ':' not as part of an object at line 1, column 37
```

If you look at the first element of the `"diagnostics"` field, it wants to treat the array as an object and uses field initialization syntax. It looks like something changed to leave off the opening and closing curly braces for the elements of the diagnostics array.

Here is the text with the error I introduced (noticed the literal term, `error`, at the end):

```fortran
module module_function_call2
    type :: softmax
    contains
      procedure :: eval_1d
    end type softmax
  contains
  
    pure function eval_1d(self, x) result(res)
      class(softmax), intent(in) :: self
      real, intent(in) :: x(:)
      real :: res(size(x))
    end function eval_1d
  
    pure function eval_1d_prime(self, x) result(res)
      class(softmax), intent(in) :: self
      real, intent(in) :: x(:)
      real :: res(size(x))
      res = self%eval_1d(x)
    end function eval_1d_prime
end module module_function_call1
error
```

**Comment by dylon** [comment]
Addressed by this PR: https://github.com/lfortran/lfortran/pull/5543

**Comment by dylon** [comment]
Closed by PR #5543
